### PR TITLE
KIM API interface update

### PR DESCRIPTION
**Pull request by ellio167**
## Purpose

provide simple access to kim api library with run-time dlload linking using a plug-in interface defined in lib/kim.

## Author(s)

Ryan S. Elliott

## Backward Compatibility

no issues

## Implementation Notes

keeps statics, except for one that can't be avoided, out of pair_kim.*

## Post Submission Checklist

_Please check the fields below as they are completed_
- [ ] The feature or features in this pull request is complete
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] One or more example input decks are included
- [ ] The source code follows the LAMMPS formatting guidelines

### CODE REVIEW

**Comment by junghans** [comment]
We need to update that in the CMake build system as well!

**Comment by akohlmey** [comment]
@ellio167 something's wrong with downloading the KIM-API source. I am getting this:
```
[akohlmey@ws18 kim]$ python3 Install.py -b
Created /home/akohlmey/compile/lammps-icms/lib/kim/Makefile.KIM_DIR
  using /home/akohlmey/compile/lammps-icms/lib/kim/installed-kim-api-v1.9.4
Downloading kim-api tarball ...
Unpacking kim-api tarball ...
Traceback (most recent call last):
  File "Install.py", line 199, in <module>
    subprocess.check_output(cmd,stderr=subprocess.STDOUT,shell=True)
  File "/usr/lib64/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/usr/lib64/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command 'cd "/home/akohlmey/compile/lammps-icms/lib/kim"; rm -rf "kim-api-v1.9.4"; tar -xJvf kim-api-v1.9.4.txz' returned non-zero exit status 2.
[akohlmey@ws18 kim]$ cat kim-api-v1.9.4.txz 
<html>
<head><title>404 Not Found</title></head>
<body bgcolor="white">
<center><h1>404 Not Found</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

**Comment by akohlmey** [comment]
there is no need to rush this. i've assigned the PR to you to signal that you are deciding when to move forward with merging this PR. please drop me a note, or add a comment here, when you are ready.

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -290,7 +290,21 @@ if(ENABLE_USER-VTK)
...
 
 if(ENABLE_KIM)
-  find_package(KIM REQUIRED)
+  find_package(KIM QUIET)
+  if(NOT KIM_FOUND)
+    message(STATUS "KIM not found - we will build our own")
+    include(ExternalProject)
+    ExternalProject_Add(kim_build
+      URL https://github.com/openkim/kim-api/archive/v1.9.3.tar.gz
+      URL_MD5 86bb67622a09bcd45332c6506dc0ddd9
```
Comment (junghans) [inline cmake/CMakeLists.txt]: @ellio167 This still needs to be changed to version 1.9.4!

File: cmake/CMakeLists.txt [hunk 2]
```diff
@@ -290,7 +290,21 @@ if(ENABLE_USER-VTK)
...
+      BUILD_IN_SOURCE 1
```
Comment (junghans) [inline cmake/CMakeLists.txt]: And a patch step is missing to configure the openkim build!

**Comment by ellio167** [comment]
OK, I've released v1.9.4; back to you.

**Comment by akohlmey** [comment]
@ellio167 had to add a little tweak to recover full `Install.py` functionality under python3.

**Pull request merged**